### DISCUSSION TITLE
[HW2] IRegarding Problem 3.4

### DISCUSSION THREAD

**Discussion by sorobedio** [seominjoon/kaist-ai605#111] [Q&A] [answered]
It is said "To make the measurement accurate, perform search many times (at least more than 1000) and take the average."
Does it mean that for each question we search the top-k index for more than 1000 times and average them?
Shouldn't the search result be the same at each run on the same query ?

**Comment by seominjoon** [comment, marked as answer]
Yes, it is the same, but if you time it with Python's `time.time()`, you will see that there will be some fluctuations due to overheads. To make an accurate measurement, perform it many times and average it.

**Reply by sorobedio** [reply]
aha okay i see now. the 1000 times average is for the precision of the execution time
 am i right?

**Reply by seominjoon** [reply]
@sorobedio Yes, you are right. it's just for a better precision.
